modules/compiler/spirv-ll/source/builder_core.cpp

```diff
@@ -1999,6 +1999,18 @@ cargo::optional<Error> Builder::create<OpFunction>(const OpFunction *op) {
       module.addName(op->IdResult(), name);
     }
 
+    // DPC++ rejects variadic functions in SYCL code, with the exception of
+    // __builtin_printf which it accepts and generates invalid SPIR-V for that
+    // calls printf, but declares printf as a non-variadic function (because
+    // SPIR-V has no variadic functions) yet calls it with the normal arguments.
+    // Patch this up.
+    if (name == "printf" && function_type->getNumParams() == 1 &&
```

Comment:
Given our discussion offline I wonder if it's worth adding a bit extra to this comment, saying that this isn't strictly legal to assume in SPIR-V, but we're making use of the fact we only really support translating C/C++/OpenCL-like languages where `printf` is a reserved symbol and we can infer its behaviour.

Reply:
Thanks, I added a comment about that. Does that look good to you or am I leaving out cases that actually we do care about?

Reply:
Thanks! Hmm actually, now you mention it, we technically support Vulkan too. I've no idea what it thinks about `printf`.

Reply:
Sorry, maybe it's more correct to say we support SPIR-V generated from GLSL, targeting the vulkan environment?

Reply:
It seems like the GLSL to SPIR-V translation changes function names which avoids the problem. Updated the comment to mention that.

Reply:
Looks good, thanks